Xamarin.PropertyEditing/ViewModels/RatioViewModel.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Xamarin.PropertyEditing.Drawing;
+using Xamarin.PropertyEditing.Resources;
+
+namespace Xamarin.PropertyEditing.ViewModels
+{
+	internal class RatioViewModel
+		: PropertyViewModel<CommonRatio>
+	{
+		public double Numerator
+		{
+			get { return Value.Numerator; }
+			set {
+				if (Value.Numerator.Equals (value))
+					return;
+
+				Value = new CommonRatio (value, Value.Denominator, Value.RatioSeparator);
+			}
+		}
+
+		public double Denominator
+		{
+			get { return Value.Denominator; }
+
+			set {
+				if (Value.Denominator.Equals (value))
+					return;
+
+				Value = new CommonRatio (Value.Numerator, value, Value.RatioSeparator);
+			}
+		}
+
+		public char RatioSeparator {
+			get { return Value.RatioSeparator; }
+
+			set {
+				if (Value.RatioSeparator.Equals (value))
+					return;
+
+				Value = new CommonRatio (Value.Numerator, Value.Denominator, value);
+			}
+		}
+
+		internal void ValueChanged (string stringValue, int caretPosition, int selectionLength, double incrementValue)
+		{
+			var separator = GetSeparator (stringValue);
+			SetRatioFromString (stringValue);
+
+			var newNumerator = Value.Numerator + incrementValue;
+			if (newNumerator < 0)
+				newNumerator = 0;
+
+			var newDenominator = Value.Denominator + incrementValue;
+			if (newDenominator < 1)
+				newDenominator = 1;
+
+			// Nothing selected, just workout out which value the caret is closest to, so we can increment it
+			if (selectionLength == 0) {
+				var sepLocation = GetSeparatorPosition (stringValue);
+				if (caretPosition <= sepLocation) {
+					Value = new CommonRatio (newNumerator, Value.Denominator, separator);
+				} else {
+					Value = new CommonRatio (Value.Numerator, newDenominator, separator);
+				}
+			} else {
+				// Increment both values.
+				Value = new CommonRatio (newNumerator, newDenominator, separator);
+			}
+		}
+
+		public string ValueString
+		{
+			get { return Value.StringValue; }
+			set {
+				if (Value.StringValue.Equals (value))
+					return;
+
+				SetRatioFromString (value);
+			}
+		}
+
+		public static char[] Separators = { ':', '/' };
+
+		public RatioViewModel (IPropertyInfo property, IEnumerable<IObjectEditor> editors)
+			: base (property, editors)
+		{
+		}
+
+		protected override void OnValueChanged ()
+		{
+			base.OnValueChanged ();
+			OnPropertyChanged (nameof (ValueString));
+		}
+
+		void SetRatioFromString (string value)
+		{
+			var strippedValue = value.Replace (" ", string.Empty);
```

Comment:
I think you could avoid both the `Replace` and the loop by splitting once on the set of admissible separators and space, with the option to remove empty entries on.

Reply:
We could probably use a bunch of tests on this parsing logic.